src/main/java/online/talkandtravel/controller/http/UserController.java

```diff
@@ -27,33 +34,28 @@
  * </ul>
  */
 @RestController
+@Log4j2
 @RequestMapping(ApiPathConstants.API_BASE_PATH + "/users")
 @RequiredArgsConstructor
 public class UserController {
   private final UserService userService;
   private final UserMapper userMapper;
+  private final AuthenticationFacade authFacade;
 
-  @Operation(description = "Update a user.")
-  @PutMapping()
-  public ResponseEntity<UserDtoWithAvatarAndPassword> update(
-      @RequestBody UserDtoWithAvatarAndPassword dto) {
-    var user = userMapper.mapToModel(dto);
-    var updatedUser = userService.update(user);
-    var userDto = userMapper.toUserDtoWithAvatarAndPassword(updatedUser);
-    return ResponseEntity.ok().body(userDto);
+  @PutMapping
+  public UpdateUserResponse update(@RequestBody @Valid UpdateUserRequest dto) {
+    return authFacade.update(dto);
   }
 
-  @Operation(description = "Get a user by ID.")
   @GetMapping("/{userId}")
-  public ResponseEntity<UserDtoWithAvatarAndPassword> findById(@PathVariable Long userId) {
+  public UserDtoBasic findById(@PathVariable @Positive Long userId) {
     var user = userService.findById(userId);
-    var userDto = userMapper.toUserDtoWithAvatarAndPassword(user);
-    return ResponseEntity.ok().body(userDto);
+    return userMapper.toUserDtoBasic(user);
   }
 
   @Operation(description = "Check if email exists.")
   @GetMapping("/exists-by-email/{email}")
```

Comment:
There's no need to change it to a request parameter, because the frontend needs to know the name of the request parameter, and the current implementation doesn't need to.

https://chatgpt.com/share/438d083f-4613-4da7-8cfe-fa8f0e40f935

Reply:
you ask incorrect question to chat gpt. Of course this endpoint is correct, but it is not the best case.
https://chatgpt.com/share/3d6c5d4e-a735-464f-9a5c-b99b4d7b5eb0